app/src/main/java/com/noalecohen/dispatcher/networking/NetworkManager.kt

```diff
@@ -0,0 +1,28 @@
+package com.noalecohen.dispatcher.networking
+
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+
+object NetworkManager {
+    var retrofit: Retrofit? = null
```

Comment:
We need to find a way that retrofit won't be nullable.
In order that the API controller will use an API that is also not nullable i.e NewsServiceApi and notNewsServiceApi?

Perhaps the base URL should not be passed but rather be taken by the NetworkManager from the BuildConfig.
In addition the application shouldn't init retrofit but instead use lazy initialization something like:
 `val retrofit : Retrofit by lazy {
    Retrofit.Builder()
        .addConverterFactory(GsonConverterFactory.create())
        .....
        .baseUrl(BuildConfig.BASE_URL)
        .build()
}
`